components/hab/src/cli.rs

```diff
@@ -264,6 +264,46 @@ pub fn get() -> App<'static, 'static> {
                     (@arg AUTH_TOKEN: -z --auth +takes_value "Authentication token for Builder")
                 )
             )
+            (@subcommand secret =>
+                (about: "Commands related to secret management")
+                (@setting ArgRequiredElseHelp)
+                (@subcommand generate => 
+                    (about: "Create and upload a secret for your origin")
+                    (@arg KEY_NAME: +required +takes_value
+                        "The name of the variable key to be injected into the studio.")
+                    (@arg SECRET: +required +takes_value
+                        "The contents of the variable to be injected into the studio.")
+                    (@arg BLDR_URL: -u --url +takes_value {valid_url}
+                        "Specify an alternate Builder endpoint. If not specified, the value will \
+                         be taken from the HAB_BLDR_URL environment variable if defined. (default: \
+                         https://bldr.habitat.sh)")
+                    (@arg AUTH_TOKEN: -z --auth +takes_value "Authentication token for Builder")
+                    (@arg ORIGIN: +takes_value
+                        "The origin for which the secret will be generated. Default is from 'HAB_ORIGIN' or cli.toml")
+                )
+                (@subcommand delete => 
+                    (about: "Delete a secret for your origin")
+                    (@arg KEY_NAME: +required +takes_value
```

Comment:
Should we add aliases to be consistent with the other commands?

Reply:
I don't think so. Personally I'd like to remove the other aliases as well. I find they often lead to typo bugs and provide an inconsistent user experience.

Reply:
I'm on-board with removing all the aliases, but I think we should do that as a separate change rather than introducing just one new command that doesn't follow the pattern and leads to an inconsistent user experience. Not a big deal; do what feels right to you.